packages/wallets/particle/src/adapter.ts

```diff
@@ -37,12 +45,37 @@ export class ParticleAdapter extends BaseMessageSignerWalletAdapter {
     private _readyState: WalletReadyState =
         typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
 
-    constructor(config: ParticleAdapterConfig = {}) {
+    private _particleNetwork: ParticleNetwork | null = null;
+
+    constructor(config: ParticleAdapterConfig) {
         super();
         this._connecting = false;
         this._publicKey = null;
         this._wallet = null;
-        this._config = config;
+
+        const nestedConfig: NestedConfig = config.config || {};
+
+        const chainId = nestedConfig.chainId !== undefined ? nestedConfig.chainId : 101;
+        const chainName = nestedConfig.chainName !== undefined ? nestedConfig.chainName : 'solana';
+
+        this._config = {
+            ...config,
+            config: {
+                ...nestedConfig,
+                chainId,
+                chainName,
+                projectId: nestedConfig.projectId,
+                clientKey: nestedConfig.clientKey,
+                appId: nestedConfig.appId,
+            },
+        };
+    }
+
+    public get auth(): ParticleNetwork['auth'] {
+        if (!this._particleNetwork) {
+            throw new Error('ParticleNetwork is not initialized.');
+        }
+        return this._particleNetwork.auth;
```

Comment:
This is to expose `.auth` on the instance of the adapter, which opens up a ton of functionality (manual logins, user info retrieval, disconnect, etc.) - `.auth.{}` is a sort of controller for the WaaS instance, thus exposing it on the instance itself allows it to be used accordingly.

Reply:
Okay, I think it might make more sense to expose the entire `particleNetwork` object (several other adapters expose this kind of thing as `adapter.wallet`).

And maybe rename the internal `_particleNetwork` as `_wallet`.

Reply:
This way the adapter is minimally aware of the wallet you want to expose, since it doesn't use it directly, and just exposes the entire thing. If you add or remove functionality later, we won't need to update the adapter code, just the SDK version.